core/server/src/main/java/alluxio/master/file/options/CreatePathOptions.java

```diff
@@ -39,6 +40,7 @@ protected CreatePathOptions() {
     mPersisted = false;
     mRecursive = false;
     mMetadataLoad = false;
+    mDefaultMode = true;
```

Comment:
would it make sense to instead of introducing a new field use `mPermission == null` to indicate that the default should be used and have CreateFileOptions define `getPermission` as follows:

```
public Permission getPermission() {
  if (mPermission != null) {
    return mPermission;
  }
  return Permission.defaults().setMode(Mode.getDefault()).applyFileUMask();
}
```

and similarly for CreateDirectoryOptions